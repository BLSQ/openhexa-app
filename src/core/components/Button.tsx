import clsx from "clsx";
import { ButtonHTMLAttributes, ReactElement } from "react";

const ButtonVariants = ["primary", "secondary", "white", "outlined", "custom"];
type ButtonVariant = typeof ButtonVariants[number];

const ButtonSizes = ["sm", "md", "lg", "xl", "xxl"];
type ButtonSize = typeof ButtonSizes[number];

export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: ButtonVariant;
  size?: ButtonSize;
  className?: string;
  leadingIcon?: ReactElement;
}

export const Classes: { [key: string]: string } = {
  base: `relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border border-transparent font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
  `,
  primary:
    "border-transparent text-white bg-blue-600 hover:bg-blue-700 focus:ring-blue-500",
  secondary:
    "border-transparent text-indigo-700 bg-indigo-100 hover:bg-indigo-200 focus:ring-indigo-500",
  white: "text-gray-800 border-gray-300 bg-white hover:bg-gray-50",
  outlined: "text-white hover:bg-white hover:text-gray-800 border-white",

  // Sizes
  sm: "text-xs px-2 py-1",
  md: "text-sm px-3 py-2.5 leading-4",
  lg: "text-sm px-4 py-2.5",
  xl: "text-base px-4 py-2",
  xxl: "text-base px-6 py-3",
};

const Button = (props: ButtonProps) => {
  const {
    variant = "primary",
    size = "md",
    className,
    children,
    leadingIcon,
    disabled,
    ...delegated
  } = props;

  const classes = clsx(
    Classes.base,
    Classes[size],
    variant !== "custom" && Classes[variant],
    className
  );
  return (
    <button className={classes} disabled={disabled} {...delegated}>
      {leadingIcon ? (
        <LeadingIcon size={size}>{leadingIcon}</LeadingIcon>
      ) : null}
      {children}
    </button>
  );
};
Button.Variants = ButtonVariants;
Button.Sizes = ButtonSizes;

const LeadingIcon = (props: { children: ReactElement; size?: ButtonSize }) => {
  return (
    <div
      className={clsx("-ml-1", props.size === ButtonSizes[0] ? "mr-1" : "mr-2")}
      aria-hidden="true"
    >
      {props.children}
    </div>
  );
};

export default Button;
