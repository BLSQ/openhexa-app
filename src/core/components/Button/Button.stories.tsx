import { DocumentArrowDownIcon } from "@heroicons/react/24/outline";
import { Story } from "@ladle/react";

import ButtonComponent, { ButtonProps } from "./Button";

interface ButtonStoryProps extends ButtonProps {
  leading: boolean;
}

export const Button: Story<ButtonStoryProps> = ({
  children,
  variant,
  size,
  leading,
}) => {
  const leadingIcon = leading ? (
    <DocumentArrowDownIcon className="h-4" />
  ) : undefined;
  return (
    <ButtonComponent size={size} variant={variant} leadingIcon={leadingIcon}>
      {children}
    </ButtonComponent>
  );
};

Button.storyName = "default";
Button.args = {
  children: "Hello world",
  leading: false,
};
Button.argTypes = {
  variant: {
    options: ButtonComponent.Variants,
    control: { type: "radio" },
    defaultValue: "primary",
  },
  size: {
    options: ButtonComponent.Sizes,
    control: { type: "select" },
    defaultValue: "md",
  },
};

export const ButtonSnapshot: Story = () => (
  <div className="space-y-6">
    {ButtonComponent.Variants.map((variant) => (
      <div key={variant} className="flex items-center gap-4">
        {ButtonComponent.Sizes.map((size) => (
          <ButtonComponent key={size} variant={variant} size={size}>
            {variant}
          </ButtonComponent>
        ))}
      </div>
    ))}
  </div>
);
ButtonSnapshot.storyName = "Snapshot";
ButtonSnapshot.meta = {
  snapshot: true,
};

const defaults = {
  title: "UI / Button",
};

export default defaults;
