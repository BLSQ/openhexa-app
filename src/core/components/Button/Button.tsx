import clsx from "clsx";
import { ButtonHTMLAttributes, ReactElement } from "react";

const ButtonVariants = [
  "primary",
  "secondary",
  "white",
  "outlined",
  "custom",
] as const;
type ButtonVariant = typeof ButtonVariants[number];

const ButtonSizes = ["sm", "md", "lg", "xl", "xxl"] as const;
type ButtonSize = typeof ButtonSizes[number];

export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: ButtonVariant;
  size?: ButtonSize;
  className?: string;
  leadingIcon?: ReactElement;
}

export const Classes: { [key: string]: string } = {
  base: `relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
  `,
  primary:
    "border-transparent text-white bg-blue-600 hover:bg-blue-700 focus:ring-blue-500",
  secondary:
    "border-transparent text-indigo-700 bg-indigo-100 hover:bg-indigo-200 focus:ring-indigo-500",
  white: "text-gray-800 border-gray-400 bg-white hover:bg-gray-50",
  outlined: "text-gray-800 border-transparent shadow-none hover:bg-gray-100",

  // Sizes
  sm: "text-xs px-2 py-1",
  md: "text-sm px-3 py-2.5 leading-4",
  lg: "text-sm px-4 py-2.5",
  xl: "text-base px-4 py-2",
  xxl: "text-base px-6 py-3",
};

const Button = (props: ButtonProps) => {
  const {
    variant = "primary",
    size = "md",
    className,
    children,
    leadingIcon,
    disabled,
    ...delegated
  } = props;

  const classes = clsx(
    Classes.base,
    Classes[size],
    variant !== "custom" && Classes[variant],
    className
  );
  return (
    <button className={classes} disabled={disabled} {...delegated}>
      {leadingIcon ? (
        <LeadingIcon size={size}>{leadingIcon}</LeadingIcon>
      ) : null}
      {children}
    </button>
  );
};
Button.Variants = ButtonVariants;
Button.Sizes = ButtonSizes;

const LeadingIcon = (props: { children: ReactElement; size?: ButtonSize }) => {
  return (
    <div
      className={clsx(
        "-ml-1",
        props.size === ButtonSizes[0] ? "mr-1" : "mr-1.5"
      )}
      aria-hidden="true"
    >
      {props.children}
    </div>
  );
};

export default Button;
