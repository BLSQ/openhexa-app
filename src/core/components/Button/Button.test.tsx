import { PlusIcon } from "@heroicons/react/24/outline";
import { render, screen, fireEvent } from "@testing-library/react";
import Button from "./Button";

describe("Button", () => {
  it("renders a Button", () => {
    render(<Button>This is a button</Button>);

    const text = screen.getByText("This is a button");
    expect(text).toBeInTheDocument();
  });

  it("calls onClick when clicked", () => {
    const onClick = jest.fn();
    render(<Button onClick={onClick}>button</Button>);

    const btn = screen.getByRole("button");
    fireEvent.click(btn);

    expect(onClick).toHaveBeenCalled();
  });

  it("does not call onClick when disabled and clicked", () => {
    const onClick = jest.fn();
    render(
      <Button disabled onClick={onClick}>
        button
      </Button>
    );

    const btn = screen.getByRole("button");
    fireEvent.click(btn);

    expect(onClick).not.toHaveBeenCalled();
  });

  it("renders all variants and all sizes of the button", () => {
    const container = render(
      <div className="space-y-6">
        {Button.Variants.map((variant) => (
          <div key={variant} className="flex items-center gap-4">
            {Button.Sizes.map((size) => (
              <Button key={size} variant={variant} size={size}>
                {variant}
              </Button>
            ))}
          </div>
        ))}
      </div>
    );

    expect(container).toMatchSnapshot();
  });

  it("renders an icon in front of the label", () => {
    const container = render(
      <Button
        disabled
        leadingIcon={<PlusIcon className="w-4" data-testid="icon" />}
      >
        button
      </Button>
    );

    const icon = screen.getByTestId("icon");

    expect(icon).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("accepts additional classes", () => {
    render(<Button className="font-black">Black</Button>);

    const button = screen.getByRole("button");

    expect(button).toHaveClass("font-black");
  });
});
