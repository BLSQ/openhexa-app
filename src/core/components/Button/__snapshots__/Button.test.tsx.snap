// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders all variants and all sizes of the button 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="space-y-6"
      >
        <div
          class="flex items-center gap-4"
        >
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-xs px-2 py-1 border-transparent text-white bg-blue-600 hover:bg-blue-700 focus:ring-blue-500"
          >
            primary
          </button>
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-sm px-3 py-2.5 leading-4 border-transparent text-white bg-blue-600 hover:bg-blue-700 focus:ring-blue-500"
          >
            primary
          </button>
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-sm px-4 py-2.5 border-transparent text-white bg-blue-600 hover:bg-blue-700 focus:ring-blue-500"
          >
            primary
          </button>
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-base px-4 py-2 border-transparent text-white bg-blue-600 hover:bg-blue-700 focus:ring-blue-500"
          >
            primary
          </button>
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-base px-6 py-3 border-transparent text-white bg-blue-600 hover:bg-blue-700 focus:ring-blue-500"
          >
            primary
          </button>
        </div>
        <div
          class="flex items-center gap-4"
        >
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-xs px-2 py-1 border-transparent text-indigo-700 bg-indigo-100 hover:bg-indigo-200 focus:ring-indigo-500"
          >
            secondary
          </button>
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-sm px-3 py-2.5 leading-4 border-transparent text-indigo-700 bg-indigo-100 hover:bg-indigo-200 focus:ring-indigo-500"
          >
            secondary
          </button>
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-sm px-4 py-2.5 border-transparent text-indigo-700 bg-indigo-100 hover:bg-indigo-200 focus:ring-indigo-500"
          >
            secondary
          </button>
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-base px-4 py-2 border-transparent text-indigo-700 bg-indigo-100 hover:bg-indigo-200 focus:ring-indigo-500"
          >
            secondary
          </button>
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-base px-6 py-3 border-transparent text-indigo-700 bg-indigo-100 hover:bg-indigo-200 focus:ring-indigo-500"
          >
            secondary
          </button>
        </div>
        <div
          class="flex items-center gap-4"
        >
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-xs px-2 py-1 text-gray-800 border-gray-400 bg-white hover:bg-gray-50"
          >
            white
          </button>
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-sm px-3 py-2.5 leading-4 text-gray-800 border-gray-400 bg-white hover:bg-gray-50"
          >
            white
          </button>
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-sm px-4 py-2.5 text-gray-800 border-gray-400 bg-white hover:bg-gray-50"
          >
            white
          </button>
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-base px-4 py-2 text-gray-800 border-gray-400 bg-white hover:bg-gray-50"
          >
            white
          </button>
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-base px-6 py-3 text-gray-800 border-gray-400 bg-white hover:bg-gray-50"
          >
            white
          </button>
        </div>
        <div
          class="flex items-center gap-4"
        >
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-xs px-2 py-1 text-gray-800 border-transparent shadow-none hover:bg-gray-100"
          >
            outlined
          </button>
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-sm px-3 py-2.5 leading-4 text-gray-800 border-transparent shadow-none hover:bg-gray-100"
          >
            outlined
          </button>
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-sm px-4 py-2.5 text-gray-800 border-transparent shadow-none hover:bg-gray-100"
          >
            outlined
          </button>
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-base px-4 py-2 text-gray-800 border-transparent shadow-none hover:bg-gray-100"
          >
            outlined
          </button>
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-base px-6 py-3 text-gray-800 border-transparent shadow-none hover:bg-gray-100"
          >
            outlined
          </button>
        </div>
        <div
          class="flex items-center gap-4"
        >
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-xs px-2 py-1"
          >
            custom
          </button>
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-sm px-3 py-2.5 leading-4"
          >
            custom
          </button>
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-sm px-4 py-2.5"
          >
            custom
          </button>
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-base px-4 py-2"
          >
            custom
          </button>
          <button
            class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-base px-6 py-3"
          >
            custom
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="space-y-6"
    >
      <div
        class="flex items-center gap-4"
      >
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-xs px-2 py-1 border-transparent text-white bg-blue-600 hover:bg-blue-700 focus:ring-blue-500"
        >
          primary
        </button>
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-sm px-3 py-2.5 leading-4 border-transparent text-white bg-blue-600 hover:bg-blue-700 focus:ring-blue-500"
        >
          primary
        </button>
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-sm px-4 py-2.5 border-transparent text-white bg-blue-600 hover:bg-blue-700 focus:ring-blue-500"
        >
          primary
        </button>
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-base px-4 py-2 border-transparent text-white bg-blue-600 hover:bg-blue-700 focus:ring-blue-500"
        >
          primary
        </button>
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-base px-6 py-3 border-transparent text-white bg-blue-600 hover:bg-blue-700 focus:ring-blue-500"
        >
          primary
        </button>
      </div>
      <div
        class="flex items-center gap-4"
      >
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-xs px-2 py-1 border-transparent text-indigo-700 bg-indigo-100 hover:bg-indigo-200 focus:ring-indigo-500"
        >
          secondary
        </button>
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-sm px-3 py-2.5 leading-4 border-transparent text-indigo-700 bg-indigo-100 hover:bg-indigo-200 focus:ring-indigo-500"
        >
          secondary
        </button>
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-sm px-4 py-2.5 border-transparent text-indigo-700 bg-indigo-100 hover:bg-indigo-200 focus:ring-indigo-500"
        >
          secondary
        </button>
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-base px-4 py-2 border-transparent text-indigo-700 bg-indigo-100 hover:bg-indigo-200 focus:ring-indigo-500"
        >
          secondary
        </button>
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-base px-6 py-3 border-transparent text-indigo-700 bg-indigo-100 hover:bg-indigo-200 focus:ring-indigo-500"
        >
          secondary
        </button>
      </div>
      <div
        class="flex items-center gap-4"
      >
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-xs px-2 py-1 text-gray-800 border-gray-400 bg-white hover:bg-gray-50"
        >
          white
        </button>
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-sm px-3 py-2.5 leading-4 text-gray-800 border-gray-400 bg-white hover:bg-gray-50"
        >
          white
        </button>
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-sm px-4 py-2.5 text-gray-800 border-gray-400 bg-white hover:bg-gray-50"
        >
          white
        </button>
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-base px-4 py-2 text-gray-800 border-gray-400 bg-white hover:bg-gray-50"
        >
          white
        </button>
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-base px-6 py-3 text-gray-800 border-gray-400 bg-white hover:bg-gray-50"
        >
          white
        </button>
      </div>
      <div
        class="flex items-center gap-4"
      >
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-xs px-2 py-1 text-gray-800 border-transparent shadow-none hover:bg-gray-100"
        >
          outlined
        </button>
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-sm px-3 py-2.5 leading-4 text-gray-800 border-transparent shadow-none hover:bg-gray-100"
        >
          outlined
        </button>
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-sm px-4 py-2.5 text-gray-800 border-transparent shadow-none hover:bg-gray-100"
        >
          outlined
        </button>
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-base px-4 py-2 text-gray-800 border-transparent shadow-none hover:bg-gray-100"
        >
          outlined
        </button>
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-base px-6 py-3 text-gray-800 border-transparent shadow-none hover:bg-gray-100"
        >
          outlined
        </button>
      </div>
      <div
        class="flex items-center gap-4"
      >
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-xs px-2 py-1"
        >
          custom
        </button>
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-sm px-3 py-2.5 leading-4"
        >
          custom
        </button>
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-sm px-4 py-2.5"
        >
          custom
        </button>
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-base px-4 py-2"
        >
          custom
        </button>
        <button
          class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-base px-6 py-3"
        >
          custom
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button renders an icon in front of the label 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-sm px-3 py-2.5 leading-4 border-transparent text-white bg-blue-600 hover:bg-blue-700 focus:ring-blue-500"
        disabled=""
      >
        <div
          aria-hidden="true"
          class="-ml-1 mr-1.5"
        >
          <svg
            aria-hidden="true"
            class="w-4"
            data-testid="icon"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 4.5v15m7.5-7.5h-15"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
        button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-sm px-3 py-2.5 leading-4 border-transparent text-white bg-blue-600 hover:bg-blue-700 focus:ring-blue-500"
      disabled=""
    >
      <div
        aria-hidden="true"
        class="-ml-1 mr-1.5"
      >
        <svg
          aria-hidden="true"
          class="w-4"
          data-testid="icon"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 4.5v15m7.5-7.5h-15"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
